typescript/src/redemption.ts

```diff
@@ -0,0 +1,299 @@
+      })
+    } else {
+      throw new Error("Redemption address must be P2PKH, P2WPKH, P2SH or P2WSH")
+    }
+  }
+
+  // If there is a change output, add it explicitly to the transaction.
+  // If we did not add this output explicitly, the bcoin library would add it
+  // anyway during funding, but if the value of the change output was very low,
+  // the library would consider it "dust" and add it to the fee rather than
+  // create a new output.
+  const changeOutputValue = mainUtxo.value - totalOutputValue - txFee
+  if (changeOutputValue > 0) {
+    transaction.addOutput({
+      script: bcoin.Script.fromAddress(walletAddress),
+      value: changeOutputValue,
+    })
+  }
+
+  await transaction.fund(inputCoins, {
+    changeAddress: walletAddress,
+    hardFee: txFee,
+    subtractFee: false,
+  })
+
+  transaction.sign(walletKeyRing)
+
+  return {
+    transactionHex: transaction.toRaw().toString("hex"),
+  }
+}
+/**
+ * Builds a length prefixed output script based on the provided address (P2PKH,
+ * P2WPKH, P2SH or P2WSH).
+ * @param address - Bitcoin address from which the output script will be build
+ * @returns The output script as a string.
+ */
+function buildOutputScript(address: string): string {
+  const script = bcoin.Script.fromAddress(address)
+  const scriptAsStr = script.toRaw().toString("hex")
+  const scriptLength = scriptAsStr.length / 2 // get length in bytes
+
+  return "0x" + scriptLength.toString(16) + scriptAsStr
+}
+
+/**
+ * Returns the Bitcoin public key in the compressed format corresponding to
+ * the provided private key.
+ * @param privateKey - The private key of in the WIF format.
+ * @returns The public key of as a string.
+ */
+function getPublicKey(privateKey: string): string {
+  const decodedPrivateKey = wif.decode(privateKey)
+
+  const keyRing = new bcoin.KeyRing({
+    privateKey: decodedPrivateKey.privateKey,
+    compressed: decodedPrivateKey.compressed,
+  })
+
+  return keyRing.getPublicKey().toString("hex")
+}
+
+/**
+ * Prepares a list of redemption requests based on the provided redeemer
+ * addresses and the wallet key. The information on the redemption requests
+ * is acquired by connecting to the provided Bridge contract interface.
+ * @dev It is up to the caller of this function to ensure that each of the
+ *      addresses represent a valid pending redemption request in the Bridge
+ *      on-chain contract. If the given address (along with the wallet public
+ *      key corresponding to the provided private key) does not represent a
+ *      valid pending redemption, this address will be skipped.
+ * @param bridge The interface to the Bridge on-chain contract.
+ * @param walletPrivateKey The private key of the wallet in the WIF format.
+ * @param redeemerAddresses The addresses that will be the recipients of the
+ *                          redeemed Bitcoins.
+ * @returns The list of redemption requests.
+ */
+async function prepareRedemptionRequests(
```

Comment:
Let's move that function between `redeemDeposits` and `createRedempionTransaction`. This will make navigation easier since it is used earlier than the functions defined above.